s3/s3_operations.cpp

```diff
@@ -245,19 +245,44 @@ namespace irods_s3 {
             }
         }
 
-
-        // if this is a replication and we're the destination, get the data size from the source dataObjInfo
+        // special treatment for replication - get data_size from source and
+        // number_of_threads from destination
         if (oprType == REPLICATE_DEST) {
+            bool found_data_size = false;
+            bool found_number_of_threads = false;
 
             for (int i = 0; i < NUM_L1_DESC; ++i) {
-
-                if (L1desc[i].inuseFlag && L1desc[i].dataObjInp && L1desc[i].dataObjInfo
-                       && L1desc[i].dataObjInp->objPath == file_obj->logical_path()
-                       && L1desc[i].dataObjInp->oprType == REPLICATE_SRC ) {
-
+                // get the data size from source dataObjInfo
+                if (!found_data_size && L1desc[i].inuseFlag && L1desc[i].dataObjInp && L1desc[i].dataObjInfo &&
+                    L1desc[i].dataObjInp->objPath == file_obj->logical_path() &&
```

Comment:
Consider doing check at the top of the loop which will `continue` if true:
```c++
if (L1desc[i].inuseFlag &&
    L1desc[i].dataObjInp &&
    L1desc[i].dataObjInfo &&
    L1desc[i].dataObjInp->objPath == file_obj->logical_path())
{
    continue;
}
```
This would simplify the two checks on 256 and 268.